src/net/metaplaylist/index.ts

```diff
@@ -0,0 +1,311 @@
+  } from "../types";
+
+import DASHTransport from "../dash";
+import SmoothTransport from "../smooth";
+
+export interface IMetaManifestInfo {
+    manifests: Array<{
+      manifest: Document;
+      url: string;
+      startTime: number;
+      endTime: number;
+      transport: "dash"|"smooth";
+    }>;
+}
+
+type transportTypes = "dash"|"smooth";
+
+const transportTypes: transportTypes[] = ["dash", "smooth"];
+
+/**
+ * Parse playlist string to JSON.
+ * Returns an array of contents.
+ * @param {string} data
+ */
+function loadMetaPlaylistData(data: string): Array<{
+    url: string;
+    startTime: number;
+    endTime: number;
+    transport: "dash"|"smooth";
+  }> {
+  const parsedMetaPlaylist = JSON.parse(data);
+  const { contents } = parsedMetaPlaylist;
+  assert(contents);
+
+  return contents;
+}
+
+/**
+ * Get transport type from segment's privateInfos
+ * @param {Object} privateInfos
+ */
+function getTypeFromPrivateInfos(privateInfos: ISegmentPrivateInfos): transportTypes {
+  const transportType =
+    transportTypes.reduce((acc: "dash"|"smooth"|undefined, val) => {
+      if (acc !== null &&
+        privateInfos[val] === null
+      ) {
+        return val;
+      }
+      return acc;
+    }, undefined);
+
+  if (!transportType) {
+    throw new Error("Undefined transport for content for metaplaylist.");
+  }
+
+  return transportType;
+}
+
+export default function(
+    options: IParserOptions = {}
+): IMetaTransportPipelines{
+
+    const transports = {
+      dash: DASHTransport(options),
+      smooth: SmoothTransport(options),
+    };
+
+    const manifestPipeline = {
+      loader(
+        { url } : IManifestLoaderArguments
+      ) : ILoaderObservable<IMetaManifestInfo> {
+        return request({
+          url,
+          responseType: "text",
+          ignoreProgressEvents: true,
+        }).mergeMap(({ value }) => {
+          const data = value.responseData;
+          const contents = loadMetaPlaylistData(data); // load data from meta document
+          const manifestsInfos$ = contents.map((content) => {
+            const transport = transports[content.transport];
+            const loader = transport.manifest.loader;
+
+            return loader({ url: content.url })
+              .filter((res) => res.type === "response")
+              .map((res) => {
+                return {
+                  // res has been filtered and is always a response event.
+                  // TS doesn't guess.
+                  manifest: (res as ILoaderResponse<Document>).value.responseData,
+                  url: content.url,
+                  startTime: content.startTime,
+                  endTime: content.endTime,
+                  transport: content.transport,
+                };
+              });
+          });
+
+          return Observable
+            .combineLatest(manifestsInfos$)
+            .map((manifestsInfos) => {
+              return {
+                type: "response" as "response",
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4325))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=178&issues=AWNyk0oMTmDVX2bfxyaz&open=AWNyk0oMTmDVX2bfxyaz)

[ISSUE_KEY]: AWNyk0oMTmDVX2bfxyaz